src/backend/distributed/commands/multi_copy.c

```diff
@@ -3596,10 +3624,22 @@ CopyGetPlacementConnection(HTAB *connectionStateHash, ShardPlacement *placement,
 	if (placement->partitionMethod == DISTRIBUTE_BY_HASH &&
 		MultiShardConnectionType != SEQUENTIAL_CONNECTION)
 	{
-		connectionFlags |= CONNECTION_PER_PLACEMENT;
+		connectionFlags |= REQUIRE_CLEAN_CONNECTION;
```

Comment:
Maybe expand the comment to capture our current knowledge:

Claiming the connection exclusively (done below) would also have the effect of opening multiple connections, but claiming the connection exclusively prevents GetConnectionIfPlacementAccessedInXact from returning the connection if it is needed for a different shard placement. By setting the REQUIRE_CLEAN_CONNECTION flag we are guaranteed to get connection that will not be returned by GetConnectionIfPlacementAccessedInXact for the remainder of the COPY, hence it safe to claim the connection exclusively. Claiming a connection exclusively prevents it from being used in other  distributed queries that happen during the COPY (e.g. if the copy logic calls a function to calculate a default value, and the function does a distributed query).